### PR TITLE
Use fend's new inline Markdown substitution API

### PR DESCRIPTION
**Pull request by printfn**
This should fix all the issues with fend's syntax inside code blocks.

### CODE REVIEW

**Review by NickGeek — CHANGES_REQUESTED** [review]

**Comment by NickGeek** [comment]
Looks good overall, just a couple small things. Thanks for looking into this!

File: app/src/app/components/note-viewer/elements/markdown/MarkdownTransformers.ts [hunk 1]
```diff
@@ -1,31 +1,37 @@
 import { ShowdownExtension } from 'showdown';
-import { evaluateFendWithTimeoutMultiple } from 'fend-wasm-web';
+import { substituteInlineFendExpressions } from 'fend-wasm-web';
 
 export type MarkdownTransformer = ShowdownExtension;
 
+interface FendResponseComponent {
```
[the remark this answers is not in the document]
Comment (printfn) [reply app/src/app/components/note-viewer/elements/markdown/MarkdownTransformers.ts:6]: No

File: app/src/app/components/note-viewer/elements/markdown/MarkdownTransformers.ts [hunk 2]
```diff
@@ -1,31 +1,37 @@
...
-			let index = 0;
-			return text.replace(/\[\[([^]+?)\]\]/gi, () => {
-				let result = results[index++];
-				if (result === 'Error: interrupted') {
-					result = 'Error: Calculation timed out';
-				}
-				if (result.startsWith('Error:')) {
-					result = `<span style="color: red"><i>${result}</i></span>`;
+			const components: FendResponseComponent[] = JSON.parse(substituteInlineFendExpressions(text, 500));
+			let result = '';
```
[the remark this answers is not in the document]
Comment (printfn) [reply app/src/app/components/note-viewer/elements/markdown/MarkdownTransformers.ts:17]: I'm appending to `result`, doesn't it need to be `let`?

[the remark this answers is not in the document]
Comment (NickGeek) [reply app/src/app/components/note-viewer/elements/markdown/MarkdownTransformers.ts:17]: Ah you do

**Review by NickGeek — APPROVED** [review]

**Pull request merged**